### ISSUE TITLE
LLM to filter irregular speech

### ISSUE THREAD

**Issue by gksoriginals** [PeoplePlusAI/sunva-ai#18]
The deaf person will also try to speak sometimes. We need a model to figure out if the transcription is poor quality or not and if it is below a particular threshold we should not show it.

**Comment by gksoriginals** [comment]
@bsbarkur thoughts on this ?

**Comment by bsbarkur** [comment]
Sure, the proper way to do this perhaps is:

1. Offer the choice of speaking OR
2. Typing

And try out both these modes above!

**Comment by gksoriginals** [comment]
Yes. In the first case we can collect some irregular speech data in privacy preserved way to train the models for them in a later stage

**Comment by bsbarkur** [comment]
https://sites.research.google/euphonia/about/
https://www.isca-archive.org/interspeech_2021/macdonald21_interspeech.pdf

References to read for irregular speech data training @gksoriginals

**Comment by gksoriginals** [comment]
@maximaminima I was thinking about a postprocessing layer using an LLM to fix the sentences.

**Comment by ambikajo** [comment]
> @maximaminima I was thinking about a postprocessing layer using an LLM to fix the sentences.

This seems like a good + quick solution. We just need to ensure that there is an option which can make sure the LLM identifies the speaker language.

**Comment by gksoriginals** [comment]
@ambikajo but we need to finetune ASR models for it to understand the irregular speech of a deaf person. I guess this is a bit complex. Speaker language identification most of the ASR models/apis do.

**Comment by ambikajo** [comment]
Ok I understand what you mean now? I can do an ASR eval.
